plugins/infinispan100/src/main/java/org/radargun/service/InfinispanRestAPI.java

```diff
@@ -23,46 +22,74 @@
  */
 public class InfinispanRestAPI {
 
+   protected final Log log = LogFactory.getLog(getClass());
+
    private ObjectMapper mapper;
-   private HttpClient httpClient;
-   private HttpResponse<String> response;
    private Integer defaultPort;
    private static final String CACHE_MANAGER_RESOURCE = "v2/cache-managers/clustered";
    private String serverIp;
-   private final Log log = LogFactory.getLog(getClass());
 
    public InfinispanRestAPI(Integer defaultPort) {
-      mapper = new ObjectMapper();
-      mapper.disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
-      mapper.setPropertyNamingStrategy(PropertyNamingStrategy.SNAKE_CASE);
-      httpClient = HttpClient.newBuilder().version(HttpClient.Version.HTTP_1_1).build();
+      this.mapper = new ObjectMapper();
+      this.mapper.disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
+      this.mapper.setPropertyNamingStrategy(PropertyNamingStrategy.SNAKE_CASE);
       this.defaultPort = defaultPort;
-      try {
-         serverIp = Inet4Address.getLocalHost().getHostAddress();
-      } catch (UnknownHostException e) {
-         log.error("Wasn't possible to get HostAddress to execute Rest Operations", e);
-         throw new RuntimeException(e);
-      }
+      this.serverIp = lookupServerIp();
    }
 
-   public CacheManagerInfo getCacheManager() {
-      String url = String.format("http://%s:%s/rest/%s", serverIp, this.defaultPort, CACHE_MANAGER_RESOURCE);
-      HttpRequest get = HttpRequest.newBuilder()
-            .uri(URI.create(url))
-            .GET()
-            .build();
+   private String lookupServerIp() {
+      // the following are predefined values in the INFINISPAN10 XML Files
+      String interfaceName = System.getProperty("radargun.infinispan.interfaceName", "public");
+      String bindAddress = System.getProperty("infinispan.bind.address", "127.0.0.1");
```

Comment:
@diegolovison I'm worried this may not work for all allowable JGroups properties. There are a couple of issues I see:

1. The JGroups XML file can use a different variable name instead of `infinispan.bind.address`
1. JGroups allows you to set the bind address value to non IP address values.

Is it possible to get this information from the Infinispan server directly? Maybe we can talk to Tristan about this?

Reply:
It is based on Infinispan. See https://github.com/infinispan/infinispan/blob/master/server/runtime/src/main/server/server/conf/infinispan.xml

Reply:
Yes, but a different XML file can use a different variable name. That's my point.